packages/mux-player/src/index.ts

```diff
@@ -103,6 +103,11 @@ function getProps(el: MuxPlayerElement, state?: any): MuxTemplateProps {
     defaultHiddenCaptions: el.defaultHiddenCaptions,
     playerSize: getPlayerSize(el),
     hasCaptions: !!getCcSubTracks(el).length,
+    // NOTE: In order to guarantee all expected metadata props are set "from the outside" when used
+    // and to guarantee they'll all be set *before* the playback id is set, using attr values here (CJP)
+    metadataVideoId: el.getAttribute(MuxVideoAttributes.METADATA_VIDEO_ID),
+    metadataVideoTitle: el.getAttribute(MuxVideoAttributes.METADATA_VIDEO_TITLE),
+    metadataViewerUserId: el.getAttribute(MuxVideoAttributes.METADATA_VIEWER_USER_ID),
```

Comment:
this could be removed and in the metadata getter of mux-player something like 

```js
    const video_id = this.getAttribute(MuxVideoAttributes.METADATA_VIDEO_ID);
    const video_title = this.getAttribute(MuxVideoAttributes.METADATA_VIDEO_TITLE);
    const viewer_user_id = this.getAttribute(MuxVideoAttributes.METADATA_VIEWER_USER_ID);

return this.video ? this.video.metadata : {
  video_id,
video_title,
viewer_user_id,
}

Reply:
Unfortunately that wouldn't work for valid use cases. Can discuss, but tl;dr - need to guarantee that the attributes for all 3 are respected when set "from the outside", even after initial video render.

Reply:
gotcha, yes that's a tricky one. great catch. 

seems because I tried to use the metadata object in template.ts because this line does pass the props `metadataVideoId` etc to the template but nothing is done with them 

```js
this.#render({ [toPropName(attrName)]: newValue });
```